File: src/Web/src/app/features/places/utils/prohibited-place-terms/catalan-prohibited-place-terms.catalog.ts
export interface ProhibitedPlaceTermsCatalog {
  readonly languageCode: string;
  readonly terms: readonly string[];
}

/** Catalan product language. Loanwords used on Catalan listings (weed, cannabis) stay here. */
export class CatalanProhibitedPlaceTermsCatalog implements ProhibitedPlaceTermsCatalog {
  readonly languageCode = 'ca';
  readonly terms = [
    'cannabis',
    'cannabic',
    'marihuana',
    'marijuana',
    'haixix',
    'hashish',
    'weed',
    'hashquarters'
  ] as const;
}

File: src/Backend/Domain/Places/ProhibitedTerms/CatalanProhibitedPlaceTermsCatalog.cs
namespace Zuppeto.Domain.Places.ProhibitedTerms;

/// <summary>
/// Catalan product language. Includes loanwords that appear on Catalan listings (weed, cannabis).
/// </summary>
public sealed class CatalanProhibitedPlaceTermsCatalog : IProhibitedPlaceTermsCatalog
{
    public string LanguageCode => "ca";

    public IReadOnlyCollection<string> Terms { get; } =
    [
        "cannabis",
        "cannabic",
        "marihuana",
        "marijuana",
        "haixix",
        "hashish",
        "weed",
        "hashquarters"
    ];
}
